src/expr/src/scalar/func.rs

```diff
@@ -3311,6 +3382,8 @@ impl UnaryFunc {
             Tan => ScalarType::Float64.nullable(in_nullable),
             Tanh => ScalarType::Float64.nullable(in_nullable),
             Cot => ScalarType::Float64.nullable(in_nullable),
+            Log10 | Ln | Exp => ScalarType::Float64.nullable(in_nullable),
+            Log10Decimal(_) | LnDecimal(_) | ExpDecimal(_) => input_type,
```

Comment:
some other functions on decimals have this scale check
```
          CeilDecimal(scale) | FloorDecimal(scale) | RoundDecimal(scale) | SqrtDec(scale) => {
                match input_type.scalar_type {
                    ScalarType::Decimal(_, s) => assert_eq!(*scale, s),
                    _ => unreachable!(),
                }
                input_type.scalar_type.nullable(in_nullable)
            }
```

not sure if thats also needed here / if we can slot these functions into that match arm